src/rating-indicator.scss

```diff
@@ -0,0 +1,206 @@
+    }
+  }
+
+  @include fd-reset();
+
+  position: relative;
+
+  &__container {
+    @include fd-fiori-focus();
+    @include fd-fiori-focus-within();
+  }
+
+  &__input {
+    @include fd-reset();
+
+    position: absolute;
+    opacity: 0.0001;
+    z-index: -1;
+    top: 3px;
+    width: $fd-rating-indicator-default-font-size;
+    height: $fd-rating-indicator-default-font-size;
+
+    &:checked ~ input + .#{$block}__label {
+      @include fd-icon-glyph('unfavorite') {
+        color: var(--sapRating_Indicator_Unselected_Icon_Color);
+      }
+
+      &:hover::before {
+        opacity: 1;
+      }
+    }
+  }
+
+  &__label {
+    @include fd-reset();
+    @include action-cursor();
+
+    @include fd-icon-glyph('favorite') {
+      font-family: "SAP-icons";
+      color: var(--sapRating_Indicator_Selected_Icon_Color);
+      font-size: $fd-rating-indicator-default-font-size;
+    }
+
+    display: block;
+    float: left;
+    margin-right: $fd-rating-indicator-default-spacing-between-icons;
+
+    @include fd-rtl() {
+      float: right;
+    }
+
+    &:hover::before {
+      opacity: var(--sapRating_Indicator_Selected_Hover_Opacity);
+      transition: opacity $fd-animation-speed ease-in;
+    }
+  }
+
+  &__dynamic-text {
+    @include fd-reset();
+    @include fd-ellipsis();
+
+    font-size: var(--sapFontSmallSize);
+    color: var(--sapContent_LabelColor);
+    margin-left: 0.5rem;
+  }
+
+  &__hide-dynamic-text {
+    > .#{$block}__dynamic-text {
+      display: none;
+    }
+  }
+
+  @include fd-disabled() {
+    pointer-events: none;
+    opacity: var(--sapRating_Indicator_Disabled_Control_Opacity);
+
+    .#{$block}__input {
+      @include fd-icon-glyph('favorite') {
+        color: var(--sapRating_Indicator_Disabled_Selected_Color);
+      }
+
+      &:checked ~ input + .#{$block}__label {
+        @include fd-icon-glyph('favorite') {
+          color: var(--sapRating_Indicator_Disabled_Unselected_Color);
+        }
+      }
+    }
+  }
+
+  @include fd-readonly() {
+    pointer-events: none;
+
+    .#{$block}__input {
+      @include fd-icon-glyph('favorite') {
+        color: var(--sapRating_Indicator_Readonly_Selected_Color);
+      }
+
+      &:checked ~ input + .#{$block}__label {
+        @include fd-icon-glyph('favorite') {
+          color: var(--sapRating_Indicator_Readonly_Unselected_Color);
+        }
+      }
+    }
+  }
+
+  @include fd-flex-center();
+
+  &.is-display-mode {
+    pointer-events: none;
+
+    .#{$block}__label {
+      @include fd-icon-glyph('favorite') {
+        color: var(--sapRating_Indicator_Display_Only_Selected_Color);
+      }
+
+      &::before {
+        font-size: 1rem;
+      }
+
+      margin-right: 0.125rem;
+    }
+
+    .#{$block}__input {
+      &:checked ~ input + .#{$block}__label {
+        @include fd-icon-glyph('favorite') {
+          color: var(--sapRating_Indicator_Display_Only_Unselected_Color);
+        }
+      }
+    }
+  }
+
+  &__with-half-star {
```

Comment:
following BEM this should be a modifier class: `fd-rating-indicator fd-rating-indicator--half-star`
`__` is for element
`--` is for modifier

Reply:
fixed